pyiceberg/catalog/hive.py

```diff
@@ -111,6 +122,13 @@
 HIVE2_COMPATIBLE = "hive.hive2-compatible"
 HIVE2_COMPATIBLE_DEFAULT = False
 
+DEFAULT_LOCK_CHECK_MIN_WAIT_TIME = 2
+DEFAULT_LOCK_CHECK_MAX_WAIT_TIME = 30
+DEFAULT_LOCK_CHECK_RETRIES = 5
+DEFAULT_LOCK_CHECK_MULTIPLIER = 2
```

Comment:
wdyt about grouping these configs into TableProperties, along with their default value

https://github.com/apache/iceberg-python/blob/7bd5d9e6c32bcc5b46993d6bfaeed50471e972ae/pyiceberg/table/__init__.py#L200

Reply:
Hi @kevinjqliu. I don't think these should be grouped in `TableProperties`. [TableProperties](https://iceberg.apache.org/docs/nightly/configuration/#reserved-table-properties) controls the behavior of a specifc table while `CatalogProperties` controls the behavior of the catalog instance. In this case, these properties controls the behavior of HiveCatalog's Lock and thus should be classified as CatalogProperties.

Currently, the convention is to put each catalog's properties in their own files. In this case, they can be in `hive.py`.
Does this sound good to you? :)

Reply:
In Iceberg-Java, these props are in TableProperties like that.

```
iceberg.hive.lock-check-min-wait-ms=xxx
iceberg.hive.lock-check-max-wait-ms=xxx
iceberg.hive.lock-timeout-ms=xxx
```

So, it is better to unify this setting?

Reply:
In java, they are in the [`HadoopConfiguration`](https://iceberg.apache.org/docs/nightly/configuration/#hadoop-configuration). The hadoop configuration for a hive catalog is set at the catalog-level:
https://github.com/apache/iceberg/blob/817a5e1be1616af77329965ac3742c14ca3ae116/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java#L630

[Metastore parse these properties from the hadoop configuration](https://github.com/apache/iceberg/blob/e10453a10d3f9237032ba7e649808347c699f6a2/hive-metastore/src/main/java/org/apache/iceberg/hive/MetastoreLock.java#L110-L125)

I call them "CatalogProperties" since pyiceberg does not have hadoop configuration. Sorry if that creates any confusion.

The only hive-lock-related config in `TableProperties` is [`engine.hive.lock-enabled`](https://github.com/apache/iceberg/blob/e7999a194dc0f32fba8fb515c9108764ed6ba6b5/core/src/main/java/org/apache/iceberg/TableProperties.java#L319) which can enable/disable lock for a single table. But that is for https://issues.apache.org/jira/browse/HIVE-26882, see the "warn" section in the end of https://iceberg.apache.org/docs/nightly/configuration/#hadoop-configuration. I don't think we need that in pyiceberg now.

WDYT?

Reply:
You are right, I have see this setting comes from  HadoopConfiguration,  "CatalogProperties" seems more suitable.

https://github.com/apache/hive/blob/1b0e9d9758a0f28c4baf7b1895bf96bf11252f73/iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/MetastoreLock.java#L62